api/v1beta1/validations.go

```diff
@@ -35,6 +35,36 @@ func validateServices(k8sClient client.Client, services []NetworkDisruptionServi
 			return fmt.Errorf("error retrieving the specified network disruption service: %w", err)
 		}
 
+		for _, port := range service.Ports {
```

Comment:
I would tend to avoid imbricated loops when it's possible to build a dictionary before instead.

I made a suggestion on network_disruption implementation that seems could be reused here in another way here.

WDYT?